fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2870,6 +2874,12 @@ public FDBMetaDataStore getMetaDataStore() {
             return metaDataStore;
         }
 
+        @Nonnull
+        public Builder setEvolvedMetaDataFileDescriptor(@Nullable Descriptors.FileDescriptor evolvedMetaDataFileDescriptor) {
```

Comment:
This method should probably be added to `BaseBuilder` as well.

Reply:
It also looks like the `evolvedMetaDataFileDescriptor` is not used within this file (i.e., it isn't passed to the meta-data builder). There should probably be a test somewhere that would have caught this.

Reply:
Hmm, interesting. FDBRecordStoreTest.open is the test that should have caught it. I debugged it step by step now, and with evolvedMetaDataFileDescriptor = null it still passes without hitting issue #3. Let me test it on top of tree master.

Reply:
I see, the test passes because of the metaDataStore.saveAndSetCurrent() call that happens right before open/uncheckedOpen calls. getRecordMetaDataAsync will return the metadata store's pre-computed recordMetaData. I will add more tests to cover this.